### PR TITLE
An enhanced/updated PR of #210 to fix messenger between scope in issue #209

### PR DESCRIPTION
**Pull request by AbdullahFaqeir**
## Issue Reference
#209 

## Description
between() scope behaves incorrectly, when asking to return the only thread between users (1, 2), it returns all the conversations which both of those users participate in.

So now, between() scope, keeps it's incorrect behaviour, yet now it's not incorrect for the case it's now supposed to do, which is returning any threads which the passed participants are in, but added a new betweenOnly() scope, which will only return the thread of which the passed participants are in.

Note : check test cases to understand more what happened.

## How To Test This?
Run unit testing

## Documentation
- Added scope betweenOnly() to get private conversations between only passed participants.
- Added test case for betweenOnly() scope.
- Code refactor.
- Fix typos.

### CODE REVIEW

**Comment by AbdullahFaqeir** [comment]
@cmgmyr I'm sorry for the nag ^^', but I need this package, if you can check the PR.

**Comment by cmgmyr** [comment]
@AbdullahFaqeir I updated the test suite to use GitHub actions as well as some other updates. Can you rebase/update and re-push your work? This should now catch any issues with MySQL and Postgres. Please let me know if you need any assistance.

File: src/Models/Thread.php [hunk 1]
```diff
@@ -195,6 +206,24 @@ public function scopeForUserWithNewMessages(Builder $query, $userId)
...
 
+    /**
+     * Returns threads between given user ids.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
+     * @param array                                 $participants
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeBetweenOnly(Builder $query, array $participants): Builder
```
Comment (cmgmyr) [inline src/Models/Thread.php:217]: Can you add some tests around this method? The PR doesn't seem to include any. Other than that, it's looking good

**Review by cmgmyr — CHANGES_REQUESTED** [review]

File: src/Models/Thread.php (hunk 1 above)
Comment (AbdullahFaqeir) [reply src/Models/Thread.php:217]: I've already wrote one, mostly it got lost when I fixed the conflict, I'll do it now.

**Comment by cmgmyr** [comment]
Perfect! I thought I saw those tests before - thanks for adding those back in and thanks so much for this PR! It's great to have this finally cleaned up!

I'll go ahead and merge this, but will wait to tag until next week. In the meantime, composer can be updated to reference `dev-master` for additional testing

**Review by cmgmyr — APPROVED** [review]

**Pull request merged**